configuration/reports/reportdescriptors/dataexports/sql/j9CaseRegistration.sql

```diff
@@ -73,47 +96,82 @@ set t.department = cna.department,
     t.commune = cna.commune,
     t.section_communal = cna.section_communal,
     t.locality = cna.locality,
-    t.street_landmark = cna.street_landmark
-;
+    t.street_landmark = cna.street_landmark;
 
+   -- identifiers
 update temp_J9_patients t
-set t.ZL_identifier =  zlEMR(t.patient_id)
-;
+set t.ZL_identifier =  zlEMR(t.patient_id);
 
 update temp_J9_patients t
-set t.Dossier_identifier =  dosId(t.patient_id)
-;
+set t.Dossier_identifier =  dosId(t.patient_id);
 
 -- patient names
 update temp_J9_patients t
     left outer join current_name_address cna on cna.person_id = t.patient_id
 set t.given_name = cna.given_name,
-    t.family_name = cna.family_name
-;
+    t.family_name = cna.family_name;
+
+-- ------------- latest mothers group -------------------------
+DROP TEMPORARY TABLE IF EXISTS temp_mg_obs;
+create temporary table temp_mg_obs 
+select o.obs_id,  o.person_id, o.concept_id, o.value_text,o.obs_datetime 
+from obs o
+inner join temp_obgyn_enc t on t.encounter_id = o.encounter_id
+where o.voided = 0
+and o.concept_id = @mothersGroup; 
+
+create index temp_mg_obs_ci on temp_mg_obs(person_id, obs_datetime);
 
--- J9 group (latest observation of entered J9 group, since program enrollment)
 update temp_J9_patients t
-set t.J9_group =  (select value_text from obs where obs_id = latestObs(t.patient_id, @mothersGroup, j9_enrollment_date))
-;
+inner join obs o on o.obs_id =
```

Comment:
Why do you have to join back on the obs table here?  Is that a limitation of mysql and temp tables?  If so, why bother adding o.value_text and o.obs_datetime to temp_mg_obs, since you don't use them?

I wonder if it would be simpler (and not much less efficient) to just create a single temp_j9_obs table with _all_ of the observations in the temp_j9_enc table, and then use that in all of the subsequent queries.

Reply:
SO yeah, a few things: Mysql doesn't allow you to reopen a temp table in the same query like you can a non-temp table.  I tested out creating a copy of the temp_mg_obs table and using that, but it was just as quick to use the actual obs table, which isn't a surprise since it is just joining on obs_id and that is indexed.  

I tested out using just a single temp_j9_obs table (which how it seems to work best in other queries) but (at least on HUMTest), it ended up being faster to create individual temp tables for each of these.  

But yeah, I can remove value_text from this.  (and similarly from the other 2 times I do this)